packages/compose/src/hooks/use-media-query/index.js

```diff
@@ -10,28 +28,30 @@ import { useState, useEffect } from '@wordpress/element';
  * @return {boolean} return value of the media query.
  */
 export default function useMediaQuery( query ) {
-	const [ match, setMatch ] = useState(
-		() =>
-			!! (
-				query &&
-				typeof window !== 'undefined' &&
-				window.matchMedia( query ).matches
-			)
+	const subscribe = useCallback(
+		/** @type {(onStoreChange: () => void) => () => void} */
+		( onStoreChange ) => {
+			const mediaQueryList = getMediaQueryList( query );
+			if ( ! mediaQueryList ) {
+				return () => {};
+			}
+
+			mediaQueryList.addEventListener( 'change', onStoreChange );
+			return () => {
+				mediaQueryList.removeEventListener( 'change', onStoreChange );
+			};
+		},
+		[ query ]
 	);
 
-	useEffect( () => {
-		if ( ! query ) {
-			return;
-		}
-		const updateMatch = () =>
-			setMatch( window.matchMedia( query ).matches );
-		updateMatch();
-		const list = window.matchMedia( query );
-		list.addListener( updateMatch );
-		return () => {
-			list.removeListener( updateMatch );
-		};
-	}, [ query ] );
+	const matches = useSyncExternalStore(
+		subscribe,
+		() => {
+			const mediaQueryList = getMediaQueryList( query );
+			return mediaQueryList?.matches ?? false;
+		},
+		() => undefined
```

Comment:
Nit: Maybe we should just return false for SSR?

Reply:
Right, it would always be `false` on the server. I wonder if the `getServerSnapshot` is needed at all. The docs say it's optional, but every implementation out in the wild adds it.

Reply:
Since it's in the hooks package, it might be used by someone else with a SSR setup. Hence I think it makes sense to add it.